app/routes/bdb/components/SemesterStatusDetail.js

```diff
@@ -73,6 +78,34 @@ export default class SemesterStatusDetail extends Component<Props, State> {
     if (!semesterStatus) return <LoadingIndicator />;
 
     const humanReadableSemester = this.semesterToHumanReadable();
+
+    const renderFile = (type: string) => {
```

Comment:
This will recreate the renderFile function on every render, it is better to put it out as a class function and passing the semesterStatus as an argument instead! Maybe split it up a bit and make it a bit more readable. 

```
const fileName = this.fileNameToShow(semesterStatus[type], ...);
const showDelete = this.state.editing && semesterStatus[type];
const showUpload = !showDelete && this.state.editing;

return (
  ...
  <span>
    {showDelete && <span className={...}><span>{fileName}</span>... }
    {showUpload && <UploadButton />}
  </span>
```
Maybe even make the whole renderFile thing as a component, to make the `{['contract', 'statistics', 'evaluation'].map(type => (` more readable?

Reply:
They're mutually exclusive though, `showDelete` and `showUpload` are never true at the same time. Doesn't that make your solution confusing? If I saw that I wouldn't assume there was an if/else relationship going on.

Reply:
I moved it to its own component though, I think that makes sense :+1:

Reply:
Didn't dive completely into when you wanted to show it and not - was just a suggestion for readability 👍🏻